protos/backend_service.proto

```diff
@@ -0,0 +1,170 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+syntax = "proto3";
+
+package durabletask.protos.backend.v1;
+
+option csharp_namespace = "Microsoft.DurableTask.Protobuf";
+option java_package = "com.microsoft.durabletask.implementation.protobuf";
+option go_package = "github.com/microsoft/durabletask-protobuf/internal/protos";
+
+import "orchestrator_service.proto";
+
+
+import "google/protobuf/timestamp.proto";
+import "google/protobuf/wrappers.proto";
+import "google/protobuf/empty.proto";
+
+// gRPC service used by Durable Task Framework (DTFx) backend implementations.
+// The RPCs in this service are used by DTFx backends to manage orchestration state.
+service BackendService {
```

Comment:
It might be good to also document the "soft" parts of this contract, such as requirements to pass the task hub name, authorization tokens, etc. as method metadata.

Reply:
That's a good question. After doing a bit of research on this topic, it seems that this wouldn't be considered best practice because proto files should be focused on defining RPCs and data structures. Things like task hubs and authorization are specific to a particular implementation of this core contract. For example, there could be other implementations that wouldn't use these concepts, such as sidecar style implementations like Dapr, or there could be implementations which use them differently.

For the purpose of maintaining separation of concerns, I recommend we leave out task hub and authorization details from the comments, and instead document those in the service/middleware layers that actually define and implement them.

Reply:
Ok, I can buy that.